### PR TITLE
Don't fail setup if sync_docs fails.

### PR DESCRIPTION
**Pull request by mvaled**
Syncing docs may fail in countries like Cuba, Iran, etc...

### CODE REVIEW

**Comment by mvaled** [comment]
The latest commits are related to what's being commented in https://forum.sentry.io/t/unable-to-update-to-lastest-master/2352

File: src/sentry/utils/distutils/commands/build_assets.py [hunk 1]
```diff
@@ -134,6 +134,8 @@ def _build_static(self):
         #   * React optimizes out certain code paths
         #   * Webpack will add version strings to built/referenced assets
         env = dict(os.environ)
+        self._run_command(['yarn'], env=env)
```
Comment (mattrobenolt) [inline src/sentry/utils/distutils/commands/build_assets.py]: Can we remove this? This seems entirely unrelated to the issue.

File: src/sentry/utils/distutils/commands/build_integration_docs.py [hunk 2]
```diff
@@ -19,4 +19,4 @@ def get_dist_paths(self):
     def _build(self):
         from sentry.utils.integrationdocs import sync_docs
         log.info('downloading integration docs')
-        sync_docs()
+        sync_docs(nice=True)
```
Comment (mattrobenolt) [inline src/sentry/utils/distutils/commands/build_integration_docs.py]: The naming `nice` is a bit awkward. I also think it's not even worth making this an argument since it's nothing else hooks into this.

I think if we want to allow this to fail, we just let it fail always.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

**Review by mvaled — COMMENTED** [review]
Hi @mattrobenolt 

I removed the commit running yarn twice.  

About the need or not of the argument.  If we let the `sync_docs` to fail entirely, and trap the exception in the distutils command, then webpack may fail with:

```
Error
    at factoryCallback (/home/manu/src/github.com/provision/sentry/node_modules/webpack/lib/Compilation.js:230:39)
    at /home/manu/src/github.com/provision/sentry/node_modules/webpack/lib/NormalModuleFactory.js:234:19
    at onDoneResolving (/home/manu/src/github.com/provision/sentry/node_modules/webpack/lib/NormalModuleFactory.js:59:20)
    at /home/manu/src/github.com/provision/sentry/node_modules/webpack/lib/NormalModuleFactory.js:126:20
    at /home/manu/src/github.com/provision/sentry/node_modules/async/dist/async.js:3861:9
    at /home/manu/src/github.com/provision/sentry/node_modules/async/dist/async.js:421:16
    at iteratorCallback (/home/manu/src/github.com/provision/sentry/node_modules/async/dist/async.js:996:13)
    at /home/manu/src/github.com/provision/sentry/node_modules/async/dist/async.js:906:16
    at /home/manu/src/github.com/provision/sentry/node_modules/async/dist/async.js:3858:13
    at /home/manu/src/github.com/provision/sentry/node_modules/webpack/lib/NormalModuleFactory.js:121:22
    at onError (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:65:10)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at runAfter (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:158:4)
    at innerCallback (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:146:3)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at next (/home/manu/src/github.com/provision/sentry/node_modules/tapable/lib/Tapable.js:234:11)
    at /home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:40:4
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at runAfter (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:158:4)
    at innerCallback (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:146:3)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at next (/home/manu/src/github.com/provision/sentry/node_modules/tapable/lib/Tapable.js:234:11)
    at innerCallback (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:144:11)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at next (/home/manu/src/github.com/provision/sentry/node_modules/tapable/lib/Tapable.js:231:35)
    at resolver.doResolve.createInnerCallback (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:44:6)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at innerCallback (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/Resolver.js:144:11)
    at loggingCallbackWrapper (/home/manu/src/github.com/provision/sentry/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at next (/home/manu/src/github.com/provision/sentry/node_modules/tapable/lib/Tapable.js:231:35)
resolve 'integration-docs-platforms' in '/home/manu/src/github.com/provision/sentry/src/sentry/static/sentry/app/views/onboarding'
  Parsed request is a module
  using description file: /home/manu/src/github.com/provision/sentry/package.json (relative path: ./src/sentry/static/sentry/app/views/onboarding)
    aliased with mapping 'integration-docs-platforms': '/home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json' to '/home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json'
      using description file: /home/manu/src/github.com/provision/sentry/package.json (relative path: ./src/sentry/static/sentry/app/views/onboarding)
        Field 'browser' doesn't contain a valid alias configuration
      after using description file: /home/manu/src/github.com/provision/sentry/package.json (relative path: ./src/sentry/static/sentry/app/views/onboarding)
        using description file: /home/manu/src/github.com/provision/sentry/package.json (relative path: ./src/sentry/integration-docs/_platforms.json)
          no extension
            Field 'browser' doesn't contain a valid alias configuration
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json doesn't exist
          .less
            Field 'browser' doesn't contain a valid alias configuration
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json.less doesn't exist
          .jsx
            Field 'browser' doesn't contain a valid alias configuration
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json.jsx doesn't exist
          .js
            Field 'browser' doesn't contain a valid alias configuration
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json.js doesn't exist
          .json
            Field 'browser' doesn't contain a valid alias configuration
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json.json doesn't exist
          as directory
            /home/manu/src/github.com/provision/sentry/src/sentry/integration-docs/_platforms.json doesn't exist
command failed [node_modules/.bin/webpack --bail] via [/home/manu/src/github.com/provision/sentry]
Traceback (most recent call last):
  File "/home/manu/src/github.com/provision/sentry/src/sentry/utils/distutils/commands/build_assets.py", line 117, in _build
    self._build_static()
  File "/home/manu/src/github.com/provision/sentry/src/sentry/utils/distutils/commands/build_assets.py", line 139, in _build_static
    self._run_command(['node_modules/.bin/webpack', '--bail'], env=env)
  File "/home/manu/src/github.com/provision/sentry/src/sentry/utils/distutils/commands/base.py", line 150, in _run_command
    return check_output(cmd, cwd=self.work_path, env=env)
  File "/usr/lib/python2.7/subprocess.py", line 574, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['node_modules/.bin/webpack', '--bail']' returned non-zero exit status 1
unable to build Sentry's static assets!
Hint: You might be running an invalid version of NPM.
```

**Comment by MaxBittker** [comment]
@mvaled might want to look at this line in the webpack config, for that stacktrace https://github.com/getsentry/sentry/blob/master/webpack.config.js#L182

**Comment by mvaled** [comment]
Hi @MaxBittker 

Unfortunately, I don't know webpack.  But I guess that this line configures a fake _platforms.json file for tests.  But I can't be sure that I should set the NODE_ENV='test' in my on-premises server.

The whole story is that sentry.io is blocked to some countries, and my server resides in one of them.  

Do you propose to change webpack config to somethig like the following?

```
'integration-docs-platforms': IS_TEST || SKIP_DOCS_PLATFORMS
        ? path.join(__dirname, 'tests/fixtures/_platforms.json')
: path.join(__dirname, 'src/sentry/integration-docs/_platforms.json')
```

and properly define the SKIP_DOCS_PLATFORMS...

**Comment by MaxBittker** [comment]
@mvaled yes, i believe that will work. sorry for this issue! another possible workaround i could imagine would be building the JS assets on a server that can access sentry.io, and then copying them to your server

**Comment by mvaled** [comment]
Hi @MaxBittker 

I updated the webpack.conf.js.  I had to keep commit 5adab41ad465d32b0068dffb6437b387a60edb74.  Otherwise the setup wouldn't reach the webpack stage.

**Comment by mvaled** [comment]
Rebased to resolve conflicts.
